### PR TITLE
CSS-10666: Add enable DB tls config

### PR DESCRIPTION
**Pull request by kelkawi-a**
This PR adds the `db-tls-enabled` config, which allows the user to specify whether or not the persistence and visibility database has TLS enabled. It also updates the integration tests and workflows in line with recent updates made to our charms.

Note: I've disabled one of the integration tests which tests that a server upgrade from v1.20 to v1.21 will result in a schema upgrade. This is due to a bug in pylibjuju in how charms are deployed from Charmhub with custom resources, which will need to be investigated later. In any case, the changes introduces here do not have any effect on this process.

### CODE REVIEW

File: tests/integration/test_auth.py
Comment (mertalpt) [inline tests/integration/test_auth.py:88]: nitpick: This is ugly. A more elegant way does not come to my mind at the moment but you can at least put the comparisons into variables and use them in the branch check.

File: tests/integration/test_server_upgrade.py
Comment (mertalpt) [inline tests/integration/test_server_upgrade.py:24]: question: What is the bug? I don't think we should disable tests to get a PR in without a very good reason.

**Review by mertalpt — COMMENTED** [review]
Will approve after threads are resolved. Re-request review when you are done.

File: tox.ini
Comment (onurmus) [inline tox.ini:104]: You may also remove some CVE issues if they are resolved with this update.

**Review by onurmus — APPROVED** [review]

File: tests/integration/test_server_upgrade.py
Comment (kelkawi-a) [reply tests/integration/test_server_upgrade.py:24]: It's related to the way pylibjuju is deploying a charm from Charmhub with custom resources. The reason this is being disabled for now is that the changes introduced here have no effect on what is being tested, so i'll add this to the description to make it clear. This test was initially written to prove that upgrading a charm from a specific version to a consecutive version will result in a schema upgrade, but does not really need to run every time.

File: tox.ini
Comment (onurmus) [reply tox.ini:104]: Wrong alert. I thought we are on temporal-worker repo 😄

File: .github/workflows/publish_charm.yaml
Comment (AmberCharitos) [inline .github/workflows/publish_charm.yaml:9]: Suggestion: this can be renamed to `publish-charm` if no testing is being conducted.

Comment (AmberCharitos) [reply .github/workflows/publish_charm.yaml:9]: Out of curiosity, what is the reason you no longer want to test on publishing of the charm? I've found this useful when there have been changes to depending charms between submitting the PR for review and the merge.

File: config.yaml
Comment (AmberCharitos) [inline config.yaml:123]: Question: is it not possible to get this information from the postgresql relation?

File: tests/unit/test_charm.py
Comment (AmberCharitos) [inline tests/unit/test_charm.py:189]: Suggestion: It might be worth having a unit test where this value is `True` to validate the logic.

**Review by AmberCharitos — COMMENTED** [review]
Code looks good, but please could you remind me why the tls-enabled is a config rather than being something deduced from the relation with postgresql and then passed to the admin charm?

File: config.yaml
Comment (kelkawi-a) [reply config.yaml:123]: You're absolutely right, I missed that the interface provides a tls flag in the relation databag. I've updated the logic to grab this flag from the relation rather than from the charm's config. Thanks!

File: .github/workflows/publish_charm.yaml
Comment (kelkawi-a) [reply .github/workflows/publish_charm.yaml]: The operator workflows changed a while back to this approach, where tests are no longer needed once a PR is merged. There's a point to be made about the need to run the test again before publishing, but I opted to use the workflow as is and adapt to any changes needed.

File: config.yaml
Comment (kelkawi-a) [reply config.yaml:123]: Coming back to this, it seems there's a bug with the postgresql-k8s charm that is causing it not to provide the `tls` flag in the relation databag. I've filed an issue [here](https://github.com/canonical/postgresql-k8s-operator/issues/698), and reverted to the config-based approach as a workaround for now.

**Pull request merged**
